src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -86,8 +86,9 @@ CEvaluationResult* CCrossValidation::evaluate()
 
 	if (m_autolock)
 	{
+		int32_t flag=0;
 		/* if machine supports locking try to do so */
-		if (m_machine->supports_locking())
+		if (m_machine->supports_locking() && flag)
```

Comment:
what does this flag do?
In general, use variable names that are self explaining :)